GhostNetwork.Messages.Api/Controllers/ChatController.cs

```diff
@@ -1,35 +1,194 @@
 ﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Mvc;
+using Swashbuckle.AspNetCore.Filters;
 
 namespace GhostNetwork.Messages.Api.Controllers
 {
     [ApiController]
     [Route("[controller]")]
     public class ChatController : ControllerBase
     {
+        private readonly IChatService _chatService;
+        private readonly IMessageService _messageService;
+
+        public ChatController(IChatService chatService, IMessageService messageService)
+        {
+            _chatService = chatService;
+            _messageService = messageService;
+        }
+
+        /// <summary>
+        /// Get exist user chats by user id
+        /// </summary>
+        /// <param name="skip">Skip exist chats up to a specified position</param>
+        /// <param name="take">Take exist chats up to a specified position</param>
+        /// <param name="userId">Filters by user</param>
+        /// <response code="200">User chat ids</response>
+        [HttpGet("search/{userId:guid}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [SwaggerResponseHeader(StatusCodes.Status200OK, "X-TotalCount", "Number", "Total number of user chats")]
+        public async Task<ActionResult> SearchExistChatsAsync(
+            [FromQuery, Range(0, int.MaxValue)] int skip,
+            [FromQuery, Range(1, 100)] int take,
+            [FromRoute] Guid userId)
+        {
+            var (existChats, totalCount) = await _chatService.SearchChatsAsync(skip, take, userId);
+
+            Response.Headers.Add("X-TotalCount", totalCount.ToString());
+            Response.Headers.Add("X-HasMore", (totalCount > skip + take).ToString());
+
+            return Ok(existChats);
```

Comment:
Returning just chat guids not really useful. I think it's better to return chat objects (id, name, participants?)